homeassistant/components/growatt_server/sensor.py

```diff
@@ -29,81 +29,263 @@
 DEFAULT_NAME = "Growatt"
 SCAN_INTERVAL = datetime.timedelta(minutes=5)
 
+# Sensor type order is: Sensor name, Unit of measurement, api data name, additional options
+
 TOTAL_SENSOR_TYPES = {
-    "total_money_today": ("Total money today", "€", "plantMoneyText", None),
-    "total_money_total": ("Money lifetime", "€", "totalMoneyText", None),
+    "total_money_today": ("Total money today", "€", "plantMoneyText", {}),
+    "total_money_total": ("Money lifetime", "€", "totalMoneyText", {}),
     "total_energy_today": (
         "Energy Today",
         ENERGY_KILO_WATT_HOUR,
         "todayEnergy",
-        "power",
+        {"device_class": "power"},
```

Comment:
Energy (kWh) should not use device class power.

https://developers.home-assistant.io/docs/core/entity/sensor#available-device-classes